FreeRTOS/FreeRTOS-Plus-TCP
# Add ipconfigEMAC_TASK_HOOK macro to STMH7xx network interface

Add ipconfigEMAC_TASK_HOOK macro to allow for configurable watchdog timer for STMH7xx MCUs

Description
-----------
- Introduced the `ipconfigEMAC_TASK_HOOK` macro, which defaults to a no-op but can be overridden in `FreeRTOSIPConfig.h`.
- Added `ipconfigEMAC_TASK_HOOK()` call in the Ethernet status checking code to enable optional watchdog timer functionality.
- This change allows users to define `ipconfigEMAC_TASK_HOOK` in `FreeRTOSIPConfig.h` to enable or disable the watchdog timer, providing greater flexibility in managing Ethernet operations.
- The macro is safely defined as an empty statement if not previously defined, ensuring backward compatibility.


Test Steps
-----------
- Ensure that FreeRTOSIPConfig.h is properly configured to include or exclude the ipconfigEMAC_TASK_HOOK macro.
- Deploy the firmware to the target hardware.
- Test Ethernet link status changes by simulating link up and link down events.
- Confirm that if ipconfigEMAC_TASK_HOOK is defined, the corresponding watchdog timer functionality is correctly invoked.

Checklist:
----------


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
